src/test/java/com/work/management/service/employee/impl/EmployeeServiceImplTest.java

```diff
@@ -48,10 +55,32 @@ public void updateEmployeeEntity_forExistingEmployee_returnsUpdatedEmployee() {
     assertThat(actualEmployeeDto).isEqualTo(employeeDto);
   }
 
-  private static EmployeeDto getEmployeeDto(Employee employee) {
-    EmployeeDto employeeDto = new EmployeeDto();
-    BeanUtils.copyProperties(employee, employeeDto);
-    return employeeDto;
+  @Test
+  public void bulkUpdate_forExistingEmployees_returnUpdatedEmployeesList() throws Exception {
+    ImmutableList<Employee> employeeList = getEmployeeList();
+    employeeList.forEach(employee -> employeeService.save(getEmployeeDto(employee)));
+
+    ImmutableList<Employee> actualBulkUpdatedEmployeesInfo = employeeService
+        .bulkUpdate(getBulkEmployeeDto());
+    ImmutableList<Employee> updatedEmployeesInfo = employeeList.stream()
+        .map(employee -> employeeService.getEmployeeByUserName(employee.getUserName()))
+        .map(EmployeeServiceImplTest::getEmployee)
+        .collect(toImmutableList());
+
+    assertThat(actualBulkUpdatedEmployeesInfo).isEqualTo(updatedEmployeesInfo);
+  }
+
+  @Test
+  public void bulkUpdate_forDuplicateEmployee_throwsBadRequestException() {
+    BadRequestException exception = assertThrows(BadRequestException.class,
+        () -> employeeService.bulkUpdate(getDuplicateBulkEmployeeDto()));
+    assertThat(exception).hasMessageThat().contains("Duplicate employee ids exist");
+  }
+
+  private static Employee getEmployee(EmployeeDto employeeDto) {
+    Employee employee = new Employee();
+    BeanUtils.copyProperties(employeeDto, employee);
+    return employee;
   }
 
```

Comment:
Add a test case when one of the employee is not found...probably need to change the implementation of bulk api for this case also.